baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py

```diff
@@ -0,0 +1,189 @@
+from baremetal_network_provisioning.ml2 import (hp_network_provisioning_driver
+                                                as np_drv)
+from baremetal_network_provisioning.ml2 import mechanism_hp as hp_mech
+
+import contextlib
+
+import mock
+from oslo_config import cfg
+
+# from neutron.extensions import portbindings
+from neutron.tests import base
+CONF = cfg.CONF
+
+
+class TestHPMechDriver(base.BaseTestCase):
+
+    def setUp(self):
+        super(TestHPMechDriver, self).setUp()
+        self.driver = hp_mech.HPMechanismDriver()
+        self.np_driver = np_drv.HPNetworkProvisioningDriver()
+        self.driver.initialize()
+        self.driver._load_drivers()
+
+    def _get_port_context(self, tenant_id, net_id, vm_id, network):
+        port = {'device_id': vm_id,
+                'device_owner': 'compute',
+                'binding:host_id': 'ubuntu1',
+                'name': 'test-port',
+                'tenant_id': tenant_id,
+                'id': 123456,
+                'network_id': net_id,
+                'binding:profile':
+                {'local_link_information': [{'switch_id': '11:22:33:44:55:66',
+                                             'port_id': 'Tengig0/1'}]},
+                'binding:vnic_type': 'baremetal',
+                'admin_state_up': True
+                }
+        return FakePortContext(port, port, network)
+
+    def _get_network_context(self, tenant_id, net_id, seg_id, shared):
+        network = {'id': net_id,
+                   'tenant_id': tenant_id,
+                   'name': 'test-net',
+                   'shared': shared}
+        network_segments = [{'segmentation_id': seg_id}]
+        return FakeNetworkContext(network, network_segments, network)
+
+    def _get_port_dict(self):
+        port_dict = {'port':
+                     {'segmentation_id': 1001,
+                      'access_type': hp_const.ACCESS,
+                      'switchports':
+                      [{'port_id': 'Tengig0/1',
+                          'switch_id': '11:22:33:44:55:66'}],
+                      'id': 123456,
+                      'is_lag': False}}
+        return port_dict
+
+    def test_create_port_precommit(self):
+        """Test create_port_precommit method."""
+        fake_port_dict = mock.Mock()
+        fake_context = mock.Mock()
+        with contextlib.nested(
+            mock.patch.object(hp_mech.HPMechanismDriver,
+                              '_is_port_of_interest',
+                              return_value=True),
+            mock.patch.object(hp_mech.HPMechanismDriver,
+                              '_construct_port',
+                              return_value=fake_port_dict),
+            mock.patch.object(np_drv.HPNetworkProvisioningDriver,
+                              'create_port',
+                              return_value=None)
+        ) as (is_port, cons_port, c_port):
+            self.driver.create_port_precommit(fake_context)
```

Comment:
how about exception cases ? when back-end driver throws exception just ensure MechanismDriver error is thrown